frontend/src/pages/modelServing/screens/global/ServeModelButton.tsx

```diff
@@ -68,7 +69,7 @@ const ServeModelButton: React.FC = () => {
           {deployButton}
         </Tooltip>
       ) : (
-        deployButton
+        !isKServeNIMEnabled && deployButton
```

Comment:
I imagine we could put this at the top of the component

```tsx
  const isKServeNIMEnabled = !!project && isProjectNIMSupported(project);
  if (isKServeNIMEnabled) {
    return null;
  }
```

In the current way you have here, you're still rendering the modals and complicating render logic that we will eventually need to support anyways.

Reply:
Updated